pkg/awsutils/awsutils.go

```diff
@@ -549,13 +587,26 @@ func (cache *EC2InstanceMetadataCache) getENIMetadata(eniMAC string) (ENIMetadat
 			PrivateIpAddress: aws.String(ip4.String()),
 		}
 	}
+	ec2ipv4Prefixes := make([]*ec2.Ipv4PrefixSpecification, 0)
+	if (eniMAC == primaryMAC && !cache.useCustomNetworking) || (eniMAC != primaryMAC) {
```

Comment:
i think we need some comments here for the condition here to help ppl understand.

Reply:
Yes will add a comment, so we had to include a flag for useCustomNetworking since we have functions like refresh SGs which shouldn't update the SG on secondary ENIs.

Reply:
just for future reference. 
Ideally high-level logic like `useCustomNetworking` shouldn't interact with low-level component like this one.
I'd prefer a function like `getENIMetadata(eniMAC string, opts... eniMetadataOpts)`, which eniMetadataOpts have options like whether to include prefix. and we have a set of default attributes to fetch. and the high-level code can pass different opts based on it's business logic(e.g. useCustomNetworking)